staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -688,31 +674,6 @@ func loggableNodeNames(nodes []*v1.Node) []string {
 	return nodeNames(nodes).List()
 }
 
-func shouldSyncUpdatedNode(oldNode, newNode *v1.Node) bool {
-	// Evaluate the individual node exclusion predicate before evaluating the
-	// compounded result of all predicates. We don't sync changes on the
-	// readiness condition for eTP:Local services or when
-	// StableLoadBalancerNodeSet is enabled, hence if a node remains NotReady
-	// and a user adds the exclusion label we will need to sync as to make sure
-	// this change is reflected correctly on ETP=local services. The sync
-	// function compares lastSyncedNodes with the new (existing) set of nodes
-	// for each service, so services which are synced with the same set of nodes
-	// should be skipped internally in the sync function. This is needed as to
-	// trigger a global sync for all services and make sure no service gets
-	// skipped due to a changing node predicate.
-	if respectsPredicates(oldNode, nodeIncludedPredicate) != respectsPredicates(newNode, nodeIncludedPredicate) {
-		return true
-	}
-	// For the same reason as above, also check for any change to the providerID
-	if oldNode.Spec.ProviderID != newNode.Spec.ProviderID {
-		return true
-	}
-	if !utilfeature.DefaultFeatureGate.Enabled(features.StableLoadBalancerNodeSet) {
-		return respectsPredicates(oldNode, allNodePredicates...) != respectsPredicates(newNode, allNodePredicates...)
```

Comment:
The main difference between "all" and "stable" predicates is node-ready.  Does this mean we won't consider readiness at all?  I can't quite wrap my head around how it will change.

Also, does this mean we can get rid of `allNodePredicates`